go/tools/releasing/README.md

```diff
@@ -0,0 +1,122 @@
+        * The stability of a given module is defined by its version in the `version.yaml` file (`v1` and above
+          is stable, `v0` is unstable).
+        * A dependency is defined by the "require" section of the module's `go.mod` file (in the current branch).
+        * A warning will be printed for each dependency of a stable module on an unstable module.
+
+## Prepare a pre-release commit
+
+Update `go.mod` for all modules to depend on the specified module set's new release.
+
+1. Run the pre-release script. It creates a branch `pre_release_<module set name>_<new version>` that will contain
+   all release changes.
+
+    ```
+    ./releasing prerelease --module-set-name <name>
+    ```
+    * The script is called with the following parameters:
+        * **module-set-name (required):** Name of module set whose version is being changed.
+          Must be listed in the module set versioning YAML.
+        * **versioning-file (optional):** Path to versioning file that contains definitions of all module sets.
+          If unspecified will default to (RepoRoot)/versions.yaml.
+        * **from-existing-branch (optional):** Name of existing branch from which to base the pre-release branch.
+          If unspecified, defaults to current branch.
+        * **skip-make (boolean flag):** Specify this flag to skip the 'make lint' ** and 'make ci' steps.
+          To be used for debugging purposes. Should not be skipped during actual release.
+
+2. Verify the changes.
+
+    ```
+    git diff main
+    ```
+
+   This should have changed the version for all modules listed in `go.mod` files to be `<new version>`.
+
+3. Update the repo's Changelog to move new changes to the new version's section.
+
+4. Push the changes to upstream and create a Pull Request on GitHub.
+   Be sure to include the curated changes from the Changelog in the description.
+
+## Tag the new release commit
+
+Once the Pull Request with all the version changes has been approved and merged, it is time to tag the merged commit.
+
+First, ensure that you have checked out the branch created in the `prerelease` step: 
+`pre_release_<module set name>_<new version>`. 
+
+1. Run the tag subcommand using the `<commit-hash>` of the commit on the main branch for the merged Pull Request.
+The tag can be given as its abbreviation, in which case the script will attempt to find and print the full SHA1 hash.
+
+    ```
+    ./releasing tag --module-set-name <name> --commit-hash <hash>
+    ```
+
+2. Push tags for ALL updated modules to the upstream remote (not your fork), which are printed by the `tag` script. 
+This step must be done manually.
+
+    ```
+    git push upstream <new tag>
+    git push upstream <new tag>
+    ...
+    ```
+   
+If you need to undo the tagging of a module set (i.e. you need to delete a module set's tags for the
+version specified in `versions.yaml`), you can run the following command.
```

Comment:
Good note; it would probably be a good idea to check that the Git Tags being deleted are not already on the upstream remote with `git ls-remote --tags`, that way no one accidentally deletes tags. Will make an update with this!